samples/client/petstore/go/go-petstore/model_pet.go

```diff
@@ -278,7 +279,9 @@ func (o *Pet) UnmarshalJSON(bytes []byte) (err error) {
 
 	varPet := _Pet{}
 
-	err = json.Unmarshal(bytes, &varPet)
+	decoder := json.NewDecoder(bytes.NewReader(data))
```

Comment:
TODO: need to add a test covering this behavior (unknown fields will throw an error here).